coops-ui/src/components/AnnualReport/Certify.vue

```diff
@@ -33,83 +37,59 @@
 </template>
 
 <script lang="ts">
-import { Component, Vue, Prop, Watch, Emit } from 'vue-property-decorator'
-import { mapState } from 'vuex'
 
-@Component({
-  computed: {
-    // Property definition for runtime environment.
-    ...mapState({ currentDate: 'currentDate' })
-  }
-})
-export default class Certify extends Vue {
-  // Local definition of computed property for static type checking.
-  // Use non-null assertion operator to allow use before assignment.
-  readonly currentDate!: string
-
-  // Model properties for the Text Field and Checkbox controls.
-  private certifyTextField: string = ''
-  private certifyCheckbox: boolean = false
+import { Component, Vue, Prop, Emit } from 'vue-property-decorator'
 
+@Component
+export default class Certify extends Vue {
   // Props passed into this component.
   @Prop({ default: '' })
   private certifiedBy: string
 
   @Prop({ default: false })
   private isCertified: boolean
 
+  @Prop({ default: '' })
+  private currentDate: string
+
   /**
-   * Computed value.
-   * @return The trimmed "Certified By" string (may be '').
+   * Lifecycle callback to always give the parent a "valid" event for its property values.
    */
-  private get trimmedCertifiedBy (): string {
-    return this.certifyTextField && this.certifyTextField.trim()
+  private created (): void {
+    this.emitValid(this.trimmedCertifiedBy && this.isCertified)
   }
 
   /**
    * Computed value.
-   * @return Whether or not this component (form) is valid.
+   * @return The trimmed "Certified By" string (may be '').
    */
-  private get isCertifyValid (): boolean {
-    return !!(this.certifyCheckbox && this.trimmedCertifiedBy)
-  }
-
-  // When prop changes, update text field.
-  @Watch('certifiedBy')
-  private onCertifiedByChanged (val): void {
-    this.certifyTextField = val
-  }
-
-  // When prop changes, update checkbox.
-  @Watch('isCertified')
-  private onIsCertifiedChanged (val): void {
-    this.certifyCheckbox = val
-  }
-
-  // When the trimmed "Certified By" string changes, signal the parent.
-  @Watch('trimmedCertifiedBy')
-  private ontrimmedCertifiedByChanged (val): void {
-    this.emitCertifiedBy(val)
-  }
-
-  // When this form's validity changes, signal the parent.
-  @Watch('isCertifyValid')
-  private onIsCertifyValidChanged (val): void {
-    this.emitIsCertified(val)
+  private get trimmedCertifiedBy (): string {
+    return this.certifiedBy && this.certifiedBy.trim()
   }
 
   // Emit an update event.
   @Emit('update:certifiedBy')
-  private emitCertifiedBy (val) {
+  private emitCertifiedBy (val: string): string {
+    this.emitValid(val && val.trim() && this.isCertified)
```

Comment:
In other places you use this.trimmedCertifiedBy and here you don't. I'd prefer to see consistency.

Reply:
trimmedCertifiedBy is the computed value for certifiedBy, but in this case val != certifiedBy yes. It has to be emitted first.